tests/Fixer/ControlStructure/YodaStyleFixerTest.php

```diff
@@ -0,0 +1,266 @@
+        );
+
+        $this->fixer->configure($config);
+        $this->doTest($expected, $input);
+
+        // test the reverse config
+        $config['yoda'] = false;
+        $this->fixer->configure($config);
+        if (null === $input) {
+            $this->doTest($expected);
+        } else {
+            $this->doTest($input, $expected);
+        }
+    }
+
+    public function provideFixCases()
+    {
+        return array(
+            // Don't fix cases.
+            array('<?php $a = 1 === 1;'),
+            array('<?php $a = $b === $c;'),
+            array('<?php $a = ((array(2))) == $a;'),
+            array('<?php $a = $a == ((array(2)));'),
+            array('<?php $a = array($a) != $a;'),
+            array('<?php $a = $a != array($a);'),
+            array('<?php $a = count($this->array[$var]) === $a;'),
+            array('<?php $a = $a === count($this->array[$var]);'),
+            array('<?php $a = ($a & self::MY_BITMASK) === $a;'),
+            array('<?php $a = $a === ($a & self::MY_BITMASK);'),
+            array('<?php $a = $this->getStuff() === $myVariable;'),
+            array('<?php $a = $myVariable === $this->getStuff();'),
+            array('<?php $a = 2 * $myVar % 3 === $a;'),
+            array('<?php return $a === 2 * $myVar % 3;'),
+            array('<?php $a = $c > 2;'),
+            // Fix cases.
+            array(
+                '<?php $z = 2 == $a; $b = $c > 1 && $c <= 10;',
+                '<?php $z = $a == 2; $b = $c > 1 && $c <= 10;',
+            ),
+            array(
+                '<?php $z = /**/2
+                 # aa
+                 == $a/***/;',
+                '<?php $z = /**/$a
+                 # aa
+                 == 2/***/;',
+            ),
+            array(
+                '<?php return 2 == ($a)?>',
+                '<?php return ($a) == 2?>',
+            ),
+            array(
+                '<?php $a = ($c === ((null === $b)));',
+                '<?php $a = ($c === (($b === null)));',
+            ),
+            array(
+                '<?php return null == $a[2];',
+                '<?php return $a[2] == null;',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index];',
+                '<?php return $this->myArray[$index] === "";',
+            ),
+            array(
+                '<?php return "" === $this->myArray[$index]->a;',
+                '<?php return $this->myArray[$index]->a === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index};',
+                '<?php return $this->myObject->{$index} === "";',
+            ),
+            array(
+                '<?php return "" === $this->myObject->{$index}->a;',
+                '<?php return $this->myObject->{$index}->a === "";',
+            ),
+            array(
+                '<?php return self::MY_CONST === self::$myVariable;',
+                '<?php return self::$myVariable === self::MY_CONST;',
```

Comment:
If by that you mean that there are some people who are going to want Yoda even when it makes no sense (when none of the operands are assignable, in this particular case), but just because they prefer to have Yoda everywhere because they like it like that then indeed, there is no issue.